hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AsyncSocket.java

```diff
@@ -118,32 +119,149 @@ public final SocketAddress remoteAddress() {
      *
      * @return the local address.
      */
-    public final SocketAddress localAddress() {
+    public final SocketAddress getLocalAddress() {
         return localAddress;
     }
 
     // TODO: This option only makes sense for blocking sockets according to StandardSocketOptions.SO_LINGER
-    public abstract void soLinger(int soLinger);
+    public abstract void setSoLinger(int soLinger);
 
-    public abstract int soLinger();
+    public abstract int getSoLinger();
 
-    public abstract void keepAlive(boolean keepAlive);
+    /**
+     * Set the SO_KEEPALIVE option.
+     *
+     * @param keepAlive a boolean indicating whether or not SO_KEEPALIVE is enabled.
+     * @throws UncheckedIOException if something failed with configuring the socket
+     */
+    public abstract void setKeepAlive(boolean keepAlive);
 
+    /**
+     * Tests if SO_KEEPALIVE is enabled.
+     *
+     * @return a boolean indicating whether or not SO_KEEPALIVE is enabled.
+     * @throws UncheckedIOException if something failed with configuring the socket
+     */
     public abstract boolean isKeepAlive();
 
-    public abstract void tcpNoDelay(boolean tcpNoDelay);
+    /**
+     * Get the interval in seconds between the last data packet sent (simple ACKs are not considered data) and
+     * the first keepalive probe; after the connection is marked to need keepalive, this counter is not used
+     * any further.
+     *
+     * @param keepAliveTime the keep alive time.
+     * @throws IllegalArgumentException if keepAliveIntvl is smaller than 0.
+     * @throws UncheckedIOException if something failed with configuring the socket
+     */
+    public abstract void setTcpKeepAliveTime(int keepAliveTime);
+
+    /**
+     * Get the interval in seconds between the last data packet sent (simple ACKs are not considered data) and
+     * the first keepalive probe; after the connection is marked to need keepalive, this counter is not used any
+     * further.
+     *
+     * @return the interval.
+     * @throws UncheckedIOException if something failed with configuring the socket
+     */
+    public abstract int getTcpKeepAliveTime();
+
+    /**
+     * Sets the interval in seconds between subsequent keepalive probes, regardless of what the connection
+     * has exchanged in the meantime
+     *
+     * @param keepaliveIntvl the interval in seconds.
+     * @throws IllegalArgumentException if keepAliveIntvl is smaller than 0.
+     * @throws UncheckedIOException if something failed with configuring the socket
+     */
+    public abstract void setTcpKeepaliveIntvl(int keepaliveIntvl);
```

Comment:
Can we mention what will happen if they use java 8 here (I mean javadoc of added 3 new options)?

Reply:
I'll add the appropriate comments (behavior undefined).